### PR TITLE
Fixing scaling and position of horizontal scrollbar when scaling left

### PR DESCRIPTION
**Pull request by MikemathuKE**

### CODE REVIEW

**Comment by Hejsil** [comment]
Hi, is there a specific issue on the issue tracker this fixes? If not could you provide a brief description of the problem fixed by this code?

**Comment by MikemathuKE** [comment]
> Hi, is there a specific issue on the issue tracker this fixes? If not could you provide a brief description of the problem fixed by this code?

Hi,
No, I do not have an issue tracker for this fix, but here is a description.
Nuklear has a scale left option, that is the scaler is placed on the left side of the window rather than the right. Whenever this option was used the window did not scale correctly; When scaling right, the window moved, which should not happen. I fixed it by removing the variable delta and used in.mouse.delta.x directly.
The second issue is, whenever the horizontal scrollbar was visible and the scaler was  placed left, the scaler and horizontal scrollbar would overlap. Unfortunately, I could not scale since the horizontal scrollbar was given priority. Therefore I fixed it by moving the position of the horizontal scrollbar so that it doesn't overlap with the scaler.

**Review by Hejsil — COMMENTED** [review]
Thanks for the detailed explanation. I've been able to reproduce the problem and confirm that this fix works. The code looks good. There is only one minor thing. Wonna have a look to @dumblob ?

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ docs/src
 *.tmp
 *.swo
 *.swp
+nuklear_docking.c
```
[the remark this answers is not in the document]
Comment (MikemathuKE) [reply .gitignore:11]: Oh yes. You are right. That is something I forgot to exclude from the pull request

[the remark this answers is not in the document]
Comment (MikemathuKE) [reply .gitignore]: I have made the correction

**Comment by dumblob** [comment]
The last but one commit from this PR seems OK but the very last one is empty for some reason - maybe `--amend` could fix it?

@RobLoach I see you went through many PRs recently. Would you be willing to join our org to increase the [bus factor](https://en.wikipedia.org/wiki/Bus_factor )? I'll send you an invitation.

Also @RobLoach could you review this PR (if you're out of time as I am, feel free to use the `compare` URL syntax on GitHub: https://github.com/Immediate-Mode-UI/Nuklear/compare/master...MikemathuKE:51a4bc8b0787b1a6cc9fa5144d256642f88d9642 but mind this is the last but one commit which might be superseeded any time and you should then compare to the whole `MikemathuKE:master` branch instead of this commit).

**Comment by RobLoach** [comment]
Moving over to the other PR at https://github.com/Immediate-Mode-UI/Nuklear/pull/371

**Pull request closed without merging**